ddtrace/tracer/tracer.go

```diff
@@ -171,9 +189,12 @@ func (t *tracer) worker() {
 			t.pushPayload(trace)
 
 		case <-ticker.C:
+			t.config.statsd.Incr("datadog.trace.flush.count", []string{"reason:scheduled"}, 1)
 			t.flushPayload()
+			t.sendSpanStats()
```

Comment:
I'm not convinced this is a good place for this. Should there be a separate goroutine periodically sending the stats, or is this sufficient?

Reply:
Try writing it as a separate goroutine and see if that "fits" better.